pkg/operator/kube_cloud_config/transform.go

```diff
@@ -0,0 +1,99 @@
+package kube_cloud_config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorclient "github.com/openshift/cluster-config-operator/pkg/operator/operatorclient"
+)
+
+func CloudConfigTransformers() map[configv1.PlatformType]cloudConfigTransformer {
+	cloudConfigTransformers := map[configv1.PlatformType]cloudConfigTransformer{
+		configv1.AWSPlatformType:   awsTransformer,
+		configv1.AzurePlatformType: azureTransformer,
+	}
+
+	return cloudConfigTransformers
+}
+
+// ValidateFile verifies a file exists, has content, and is a regular file
+func ValidateFile(path string) error {
+	st, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	if !st.Mode().IsRegular() {
+		return fmt.Errorf("%s is not a regular file.", path)
+	}
+	if st.Size() <= 0 {
+		return fmt.Errorf("%s is empty.", path)
+	}
+	return nil
+}
+
+func BootstrapTransformer(infrastructureFile string, cloudProviderFile string) ([]byte, error) {
+
+	// Read, parse, and save the infrastructure object
+	var clusterInfrastructure configv1.Infrastructure
+	fileData, err := ioutil.ReadFile(infrastructureFile)
+	if err != nil {
+		return nil, err
+	}
+	if len(fileData) > 0 {
+		err := yaml.Unmarshal(fileData, &clusterInfrastructure)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Read, parse, and save the user provided cloud configmap
+	var cloudProviderConfigInput corev1.ConfigMap
+	if cloudProviderFile != "" {
+		fileData, err = ioutil.ReadFile(cloudProviderFile)
+		if err != nil {
+			return nil, err
+		}
+		if len(fileData) > 0 {
+			err := yaml.Unmarshal(fileData, &cloudProviderConfigInput)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// Determine the platform type
+	var platformName configv1.PlatformType
+	if pstatus := clusterInfrastructure.Status.PlatformStatus; pstatus != nil {
+		platformName = pstatus.Type
+	}
+	if len(platformName) == 0 {
+		platformName = clusterInfrastructure.Status.Platform
+	}
+
+	// Determine the platform specific transformer method to use
+	cloudConfigTransformers := CloudConfigTransformers()
+	cloudConfigTransformerFn, ok := cloudConfigTransformers[platformName]
+	if !ok {
+		cloudConfigTransformerFn = asIsTransformer
+	}
```

Comment:
maybe all this can be part of the `CloudConfigTransformers` that takes the infratructure object and returns the transformer.. could also be used by sync.